File: src/main/java/com/yu/iotplatform/entity/Device.java
package com.yu.iotplatform.entity;

import com.baomidou.mybatisplus.annotation.*;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("`iot_device`")
public class Device {

    @TableId(type = IdType.AUTO)
    private Long id;

    @TableField("device_id")
    private String deviceId;
    private String deviceName;
    private String deviceType;
    private String firmwareVersion;
    private String ipAddress;
    private String macAddress;
    private String location;
    private Long ownerId;
    private String status;

    private LocalDateTime lastActiveTime;

    @TableField(value = "created_at",fill = FieldFill.INSERT)
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime createdAt;

    @TableField(value = "updated_at", fill = FieldFill.UPDATE)
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime updatedAt;
}

File: entity/device.go
package entity

import (
	"github.com/yu/iot-platform-go/common"
)

// Device 设备实体，对应表 iot_device
type Device struct {
	ID              uint            `gorm:"primaryKey;autoIncrement" json:"id"`
	DeviceID        string          `gorm:"column:device_id;size:50;uniqueIndex;not null" json:"deviceId"`
	DeviceName      string          `gorm:"size:100" json:"deviceName"`
	DeviceType      string          `gorm:"size:50" json:"deviceType"`
	FirmwareVersion string          `gorm:"size:50" json:"firmwareVersion"`
	IPAddress       string          `gorm:"size:45" json:"ipAddress"`
	MacAddress      string          `gorm:"size:17" json:"macAddress"`
	Location        string          `gorm:"size:255" json:"location"`
	OwnerID         uint            `json:"ownerId"`
	Status          string          `gorm:"size:50" json:"status"`
	LastActiveTime  common.DateTime `gorm:"column:last_active_time" json:"lastActiveTime"`
	CreatedAt       common.DateTime `gorm:"column:created_at" json:"createdAt"`
	UpdatedAt       common.DateTime `gorm:"column:updated_at" json:"updatedAt"`
}

// TableName 指定表名
func (Device) TableName() string {
	return "iot_device"
}

// DeviceStatus 设备状态常量
const (
	DeviceStatusOnline  = "ONLINE"
	DeviceStatusOffline = "OFFLINE"
)
